apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java

```diff
@@ -110,10 +111,13 @@
     @Getter(AccessLevel.PACKAGE)
     private final ExtensionContext extensionContext;
 
+    //CDS watcher
+    private final SpanLimitWatcher spanLimitWatcher;
+
     /**
      * Initialize all fields with default value.
      */
-    TracingContext(String firstOPName) {
+    TracingContext(String firstOPName, SpanLimitWatcher spanLimitWatcher) {
```

Comment:
If number of parameters keeps growing, we will need to add `TracingContextConfig` as the constructor parameter, rather than keeping adding. For now, it is fine to keep two.

Reply:
Yes, I see, We can add `TracingContextConfig ` in the PR later.